contents/blog/session-recording-performance.md

```diff
@@ -0,0 +1,182 @@
+
+**Simple Load: Commandbar**
+
+![simple load commandbar](../images/blog/session-recording-performance/Screen_Shot_2022-08-30_at_11.07.36_PM.png)
+
+**Simple Load: Explo** 
+
+![simple load explo](../images/blog/session-recording-performance/Screen_Shot_2022-08-31_at_12.11.19_AM.png)
+
+> **Shallow Side** is an object’s size. 
+>
+> **Retained Size** measures the cumulative size of an object and all objects it actively references and is more relevant when measuring an object’s heap impact.
+
+Quite obviously, MutationObserver made near-zero impact. For reference, Commandbar’s total heap was over 60,000,000 bytes (60MB) large. That makes MutationRecord’s footprint a … **+0.00043%** addition to memory. **Nothing**. 
+
+### Memory: Long Term
+
+Now, let’s examine rolling use. Remember, session recordings grow with time. I re-ran each trial but with 120 seconds of active use per product. The results were … *drumroll* … the same. Roughly, at least. 
+
+**Long Trial: Brevy** 
+
+![long trial brevy](../images/blog/session-recording-performance/Screen_Shot_2022-08-30_at_10.32.07_PM.png)
+
+**Long Trial: Commandbar**
+
+![long trial commandbar](../images/blog/session-recording-performance/Screen_Shot_2022-08-30_at_11.11.28_PM.png)
+
+**Long Trial: Explo**
+
+![long trial explo](../images/blog/session-recording-performance/Screen_Shot_2022-08-31_at_12.12.29_AM.png)
+
+If you dig into the numbers, you’ll notice that there was a slight increase across the board. But just *slight*. Why? The deltas for MutationObserver are small, compressed representations of changes in the DOM tree. Even with heavy use, the allotted space barely moves. MutationObserver remains in zero-point-zero-zero territory. 
+
+> **Verdict:** MutationObserver has **no** discernible impact to page memory.
+
+### Network: Transmission
+
+Network is the bigger metric to interrogate. After all, computers have gotten faster, memory has gotten cheaper, but bad internet is still… pretty damn common. How badly does SessionRecording impact your internet speed? 
+
+Let’s take a look at the Network tab on all three apps. 
+
+**Network Tab: Brevy**
+
+![Average Packet Size: 310 bytes](../images/blog/session-recording-performance/Screen_Shot_2022-08-31_at_1.45.10_AM.png)
+
+Average Packet Size: 310 bytes
+
+**Network Tab: Commandbar:** 
+
+![Average Packet Size: 309 bytes](../images/blog/session-recording-performance/Screen_Shot_2022-08-30_at_11.14.03_PM.png)
+
+Average Packet Size: 309 bytes
+
+**Network Tab: Explo:** 
+
+![Average Packet Size: 309 bytes](../images/blog/session-recording-performance/Screen_Shot_2022-08-31_at_12.12.48_AM.png)
+
+Average Packet Size: 309 bytes
+
+All of the gzip-compressed transmissions? That’s **rrweb** / **PostHog** relaying the mutation data to a PostHog host for storage. 
+
+You will notice two things. 
+
+- These packets are **tiny** in comparison to anything else—300 or so bytes. Bytes, not kilobytes. That’s it!
```

Comment:
Of note, the 309 bytes is the size of the response payload, not the size of the uploaded data (which is significantly larger)

Reply:
Thanks Rick, I'll get a PR up to update the live article. Apologies for the rush! Damn newsletters!